transport/grpc_test.go

```diff
@@ -95,23 +103,99 @@ func TestGRPCConstructor(t *testing.T) {
 
 func TestGRPCSuccess(t *testing.T) {
 	doWithGRPCTestEnv(t, "example-caller", 5, []transport.Procedure{
-		newTestJSONProcedure("example", "Foo::Bar", testBar),
-	}, func(t *testing.T, grpcTestEnv *grpcTestEnv) {
-		request, err := newTestJSONRequest("example", "Foo::Bar", &testBarRequest{One: "hello"})
-		require.NoError(t, err)
-		response, err := grpcTestEnv.Transport.Call(context.Background(), request)
-		require.NoError(t, err)
-		require.NotNil(t, response)
-		testBarResponse := &testBarResponse{}
-		require.NoError(t, json.Unmarshal(response.Body, testBarResponse))
-		require.Equal(t, "hello", testBarResponse.One)
-	}, 0)
+		newTestJSONProcedure("example", "Foo::Bar", testBar)}, "json",
+		func(t *testing.T, grpcTestEnv *grpcTestEnv) {
+			request, err := newTestJSONRequest("example", "Foo::Bar", &testBarRequest{One: "hello"})
+			require.NoError(t, err)
+			response, err := grpcTestEnv.Transport.Call(context.Background(), request)
+			require.NoError(t, err)
+			require.NotNil(t, response)
+			testBarResponse := &testBarResponse{}
+			require.NoError(t, json.Unmarshal(response.Body, testBarResponse))
+			require.Equal(t, "hello", testBarResponse.One)
+		}, 0)
+}
+
+type simpleSvc struct {
+	streamsOpened int
+}
+
+func (s *simpleSvc) Baz(c context.Context, in *simple.Foo) (*simple.Foo, error) {
+	return in, nil
+}
+
+func (s *simpleSvc) BidiStream(stream simple.Bar_BidiStreamServer) error {
+	s.streamsOpened++
+	for {
+		msg, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		stream.Send(msg)
+	}
+	return nil
+}
+
+func (s *simpleSvc) ClientStream(simple.Bar_ClientStreamServer) error {
+	return nil
+}
+
+func (s *simpleSvc) ServerStream(*simple.Foo, simple.Bar_ServerStreamServer) error {
+	return nil
+}
+
+func TestGRPCStream(t *testing.T) {
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	assert.NoError(t, err)
+
+	server := googleGrpc.NewServer()
+	svc := &simpleSvc{}
+	simple.RegisterBarServer(server, svc)
+	go func() {
+		assert.NoError(t, server.Serve(lis))
+	}()
+	defer server.Stop()
+
+	client, err := NewGRPC(GRPCOptions{
+		Addresses: []string{lis.Addr().String()},
+		Tracer:    opentracing.NoopTracer{},
+		Caller:    "test",
+		Encoding:  "proto",
+	})
+	assert.NoError(t, err)
+
+	streamClient, ok := client.(StreamTransport)
+	assert.True(t, ok)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
is `time.Minute` too long? Don't want make the CI/local test too long

Reply:
Ah my bad, didn't replace it with second after I finished local debugging